### DISCUSSION TITLE
Do I need a signing key for an androidapp with AIR?

### DISCUSSION THREAD

**Discussion by kanisuki** [airsdk/Adobe-Runtime-Support#2764] [General]
Hello.
I use AIR and I make certificate with AIR.
Do I need the signing key? (AppBundle813-1.jpg)
If I need it, do I use the Google-generated key?(AppBundle813-2.jpg).
![AppBundle813-1jpg](https://github.com/airsdk/Adobe-Runtime-Support/assets/61077580/2262c788-84b7-4742-9f61-8c310719c0dd)
![AppBundle813-2](https://github.com/airsdk/Adobe-Runtime-Support/assets/61077580/6bf1468d-1732-4d61-bb8b-a7cf156084fa)

**Comment by balaedit** [comment]
I used a google-generated key. @marchbold and @ajwfrost can give proper answer to this in this AIR discussion.

**Reply by kanisuki** [reply]
Thanks.

**Comment by marchbold** [comment]
Yes you need a certificate to sign your application. The needs are dependent on the store you are deploying too but I'll speak about Google Play here.

You need to sign your application. User's when downloading your app will need a signed version to verify it's authenticity.
With AABs (which consist of a package of multiple architectures) Google will construct an APK from your AAB specific for a user's device architecture, sign it with your app certificate and deliver it to the user.

This app certificate is stored by the Play Store and you don't have to worry about losing it.   

How this certificate is generated is what is being asked in the screenshot. You can either allow Google to construct one, in which case you won't be able to sign other builds with it, or you can construct one yourself and upload it. It doesn't really matter which approach you take from AIR's point of view, however if you have any other needs to have access to the certificate (such as distributing to other stores) you may wish to generate your own. Generally we suggest using different certificates for different stores however there are scenarios where you may wish to use the same (eg using Google services when distributing through the Samsung store). So this choice really comes down to your particular needs, letting Google generate it is simpler, however having access to the certificate is more flexible but less secure.

Once you have decided this you can have a read over the details on signing here: 

https://airsdk.dev/docs/tutorials/platform/android/packaging-android-app-bundles#signing-and-certificates

**Reply by kanisuki** [reply]
Thank you. I will use Google-generated key.
